homeassistant/components/plaato/__init__.py

```diff
@@ -60,31 +85,126 @@
 )
 
 
-async def async_setup(hass, hass_config):
+async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Plaato component."""
+    hass.data.setdefault(DOMAIN, {})
     return True
 
 
-async def async_setup_entry(hass, entry):
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Configure based on config entry."""
-    if DOMAIN not in hass.data:
-        hass.data[DOMAIN] = {}
 
-    webhook_id = entry.data[CONF_WEBHOOK_ID]
-    hass.components.webhook.async_register(DOMAIN, "Plaato", webhook_id, handle_webhook)
+    use_webhook = entry.data.get(CONF_USE_WEBHOOK, False)
+
+    if use_webhook:
+        setup_webhook(hass, entry)
+    else:
+        await async_setup_coordinator(hass, entry)
 
-    hass.async_create_task(hass.config_entries.async_forward_entry_setup(entry, SENSOR))
+    for platform in PLATFORMS:
+        if entry.options.get(platform, True):
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(entry, platform)
+            )
 
     return True
 
 
-async def async_unload_entry(hass, entry):
+def setup_webhook(hass: HomeAssistant, entry: ConfigEntry):
+    """Init webhook based on config entry."""
+    if entry.data[CONF_WEBHOOK_ID] is None:
+        raise InvalidStateError
```

Comment:
How is this possible? What should the user do if this happens?

Reply:
It should always be set yes. Can't see a case when it shouldn't. Good catch!